### PR TITLE
swarm: migrated overlay sim to network rewrite

### PR DESCRIPTION
**Pull request by holisticode**
swarm: migrated overlay sim to rewrite

### CODE REVIEW

File: p2p/simulations/http.go [hunk 1]
```diff
@@ -455,9 +463,21 @@ func (s *Server) StreamNetworkEvents(w http.ResponseWriter, req *http.Request) {
...
+		throttle = time.Tick(rate)
+	}
+
 	for {
 		select {
 		case event := <-events:
+			//if throttling, wait for the next throttle tick before emitting event
+			if throttling {
+				<-throttle
+			}
```
Comment (gbalint) [inline p2p/simulations/http.go]: Shouldn't this throttling code should be before the select? If there is no event the Tick will just flood the throttle channel without anything reading it.

File: p2p/simulations/http.go [hunk 2]
```diff
@@ -455,9 +463,21 @@ func (s *Server) StreamNetworkEvents(w http.ResponseWriter, req *http.Request) {
 		}
 	}
 
+	//Only do throttling if requested
+	if req.URL.Query().Get("throttling") == "true" {
+		throttling = true
+		//rate at which we allow events to be emitted on the server-side event stream
+		rate := time.Second / 40
+		throttle = time.Tick(rate)
```
Comment (janos) [inline p2p/simulations/http.go]: I am suggesting to use time.NewTicker and to defer the Stop method to release the resources that the ticker acquires.

**Comment by acud** [comment]
we got this while running the simulation

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x45790b3]

goroutine 31773 [running]:
github.com/ethereum/go-ethereum/swarm/network.(*Kademlia).Healthy(0xc420a80140, 0x0, 0x0)
	/Users/stolen/go/src/github.com/ethereum/go-ethereum/swarm/network/kademlia.go:681 +0x83
github.com/ethereum/go-ethereum/swarm/network.(*Hive).Healthy(0xc4203a8a00, 0x0, 0x0)
	<autogenerated>:1 +0x47
reflect.Value.call(0xc42045b6e0, 0xc420ac0bd0, 0x13, 0x471a710, 0x4, 0xc421071da0, 0x2, 0x2, 0x1, 0xc421071da0, ...)
	/usr/local/Cellar/go/1.9.4/libexec/src/reflect/value.go:434 +0x905
reflect.Value.Call(0xc42045b6e0, 0xc420ac0bd0, 0x13, 0xc421071da0, 0x2, 0x2, 0x1, 0x1, 0x199)
	/usr/local/Cellar/go/1.9.4/libexec/src/reflect/value.go:302 +0xa4
github.com/ethereum/go-ethereum/rpc.(*Server).handle(0xc420b792a0, 0x4ab1f00, 0xc420d03d70, 0x4ab59a0, 0xc42092cf50, 0xc420fd8420, 0xc4207e99c0, 0xc420e6afa0, 0x4462123)
	/Users/stolen/go/src/github.com/ethereum/go-ethereum/rpc/server.go:311 +0x6ce
github.com/ethereum/go-ethereum/rpc.(*Server).exec(0xc420b792a0, 0x4ab1f00, 0xc420d03d70, 0x4ab59a0, 0xc42092cf50, 0xc420fd8420)
	/Users/stolen/go/src/github.com/ethereum/go-ethereum/rpc/server.go:333 +0x1ba
github.com/ethereum/go-ethereum/rpc.(*Server).serveRequest.func2(0xc420689d30, 0xc420b792a0, 0xc4204b1c90, 0x4ab59a0, 0xc42092cf50, 0xc420271718, 0x1, 0x1, 0x46fb200)
	/Users/stolen/go/src/github.com/ethereum/go-ethereum/rpc/server.go:206 +0xad
created by github.com/ethereum/go-ethereum/rpc.(*Server).serveRequest
	/Users/stolen/go/src/github.com/ethereum/go-ethereum/rpc/server.go:201 +0x2ab
exit status 2
```

**Comment by holisticode** [comment]
I actually don't think the stack trace relates to this overlay simulation. The overlay simulation at no point calls the `Healthy` RPC call which would trigger this situation.

**Review by zelig — APPROVED** [review]

**Review by gbalint — APPROVED** [review]

**Pull request merged**